azurelinuxagent/common/osutil/redhat.py

```diff
@@ -140,6 +139,30 @@ def get_nm_controlled(self):
 
         return True
 
+    def check_and_recover_nic_state(self, ifname):
+        filepath = "/sys/class/net/{0}/operstate".format(ifname)
+        try:
+            if not os.path.isfile(filepath):
+                logger.warn("Unable to determine primary network interface state, because state file does not exist: "
+                            "{0}".format(filepath))
+            else:
+                content = fileutil.read_file(filepath).rstrip().encode('utf-8')
+                down_states = ["down", "unknown"]
+                if content in down_states:
+                    logger.warn("The primary network interface {0} is in '{1}' state after being restarted. Restarting "
+                                "the Network Manager service to recover the network interface".format(ifname, content))
+                    self.restart_network_manager()
```

Comment:
after restarting the nm, we should log/send telemetry the success/failure of the attempt to bring the iface up

Reply:
Updated this. In my testing I saw that it takes around 1 second after restarting the NM for the interface to come up, so I added a 1 second sleep between NM restart and checking if the attempt succeeded